src/daemon/control.rs

```diff
@@ -886,6 +886,159 @@ pub fn handle_rpc_messages(
+                                    ),
+                                    vault.derivation_index,
+                                ));
+                            }
+                            status => {
+                                response_tx.send(Err(RpcControlError::InvalidStatus((
+                                    status,
+                                    VaultStatus::Active,
+                                ))))?;
+                                break;
+                            }
+                        },
+                        None => {
+                            response_tx.send(Err(RpcControlError::UnknownOutpoint(*outpoint)))?;
+                            break;
+                        }
+                    }
+                }
+                if txins.len() != outpoints.len() {
+                    // There was an error with one of the vaults, error is already sent.
+                    continue;
+                }
+
+                // Mutable as we *may* add a change output
+                let mut txos: Vec<SpendTxOut> = destinations
+                    .into_iter()
+                    .map(|(addr, value)| {
+                        let script_pubkey = addr.script_pubkey();
+                        SpendTxOut::Destination(ExternalTxOut::new(TxOut {
+                            value,
+                            script_pubkey,
+                        }))
+                    })
+                    .collect();
+
+                log::debug!(
+                    "Creating a Spend transaction with txins: '{:?}' and txos: '{:?}'",
+                    &txins,
+                    &txos
+                );
+
+                // This adds the CPFP output so create a dummy one to accurately compute the
+                // feerate.
+                let nochange_tx = match spend_tx_from_deposits(
+                    txins.clone(),
+                    txos.clone(),
+                    &revaultd.unvault_descriptor,
+                    &revaultd.cpfp_descriptor,
+                    xpub_ctx,
+                    revaultd.unvault_csv,
+                    revaultd.lock_time,
+                    /* Deactivate insane feerate check */
+                    false,
+                ) {
+                    Ok(tx) => tx,
+                    Err(e) => {
+                        // Why doesn't the compiler recursively handle into()s ?
+                        response_tx.send(Err(RpcControlError::Transaction(e.into())))?;
+                        continue;
+                    }
+                };
+
+                // If the feerate of the transaction would be much lower (< 90/100) than what they
+                // requested for, tell them.
+                let nochange_feerate_vb = assume_some!(
+                    nochange_tx.max_feerate().checked_mul(4),
+                    "bug in feerate computation"
+                );
+                if nochange_feerate_vb * 10 < feerate_vb * 9 {
```

Comment:
add comment explaining why *10 and *9

Reply:
There is a comment just above explaining this:
```rust
                // If the feerate of the transaction would be much lower (< 90/100) than what they
                // requested for, tell them.
```